src/Strategy/Swift/Xcode/PbxprojParser.hs

```diff
@@ -0,0 +1,187 @@
+  noneOf,
+  sepEndBy,
+  some,
+  (<?>),
+  (<|>),
+ )
+import Text.Megaparsec.Char (char, string)
+import Text.Megaparsec.Char.Lexer qualified as Lexer
+
+type Parser = Parsec Void Text
+
+sc :: Parser ()
+sc =
+  Lexer.space
+    (void $ some $ char ' ' <|> char '\t' <|> char '\n' <|> char '\r')
+    (Lexer.skipLineComment "//")
+    (Lexer.skipBlockComment "/*" "*/")
+
+lexeme :: Parser a -> Parser a
+lexeme = Lexer.lexeme sc
+
+symbol :: Text -> Parser Text
+symbol = Lexer.symbol sc
+
+betweenCurlyBrackets :: Parser a -> Parser a
+betweenCurlyBrackets = between (symbol "{") (symbol "}")
+
+betweenParentheses :: Parser a -> Parser a
+betweenParentheses = between (symbol "(") (symbol ")")
+
+parseQuotedText :: Parser Text
+parseQuotedText = between (symbol "\"") (symbol "\"") quoted
+  where
+    quoted :: Parser Text
+    quoted = toText <$> many (nullifiedQuote <|> notEscapedQuote)
+
+    nullifiedQuote :: Parser Char
+    nullifiedQuote = string "\\\"" >> pure '"'
+
+    notEscapedQuote :: Parser Char
+    notEscapedQuote = noneOf ['\"']
+
+parseText :: Parser Text
+parseText = takeWhile1P (Just "text") (\c -> c `notElem` [';', ',', ')', ' ', '\t', '\n', '\r'])
+
+-- | Potential type represented in Ascii plist file.
+-- Reference : https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/PropertyLists/OldStylePlists/OldStylePLists.html
+data AsciiValue
+  = -- | Represents SomeText or "SomeText"
+    -- Since we are only interested in textual representation of package name, and package version
+    -- We represent potential binary, date, boolean ascii type as text.
+    AText Text
+  | -- | Represents {key = value;}
+    ADict (Map Text AsciiValue)
+  | -- | Represents (A, B,)
+    AList [AsciiValue]
+  deriving (Show, Eq, Ord)
+
+data AsciiKeyValue = AsciiKeyValue Text AsciiValue deriving (Show, Eq, Ord)
+
+parseAsciiText :: Parser AsciiValue
+parseAsciiText = AText <$> lexeme (try parseQuotedText <|> parseText)
+
+parseAsciiList :: Parser AsciiValue
+parseAsciiList = AList <$> betweenParentheses (sepEndBy parseAsciiValue (symbol ","))
+
+parseAsciiValue :: Parser AsciiValue
+parseAsciiValue = try parseAsciiDict <|> try parseAsciiList <|> parseAsciiText
+
+parseAsciiDict :: Parser AsciiValue
+parseAsciiDict = ADict <$> (Map.fromList <$> lexeme (betweenCurlyBrackets $ sepEndBy (try parseAsciiKeyValue) (symbol ";")))
+
+parseAsciiKeyValue :: Parser (Text, AsciiValue)
+parseAsciiKeyValue = do
+  key <- lexeme parseText <* symbol "="
+  value <- lexeme $ try parseAsciiList <|> try parseAsciiDict <|> parseAsciiText
+  pure (key, value)
+
+-- | Represents Xcode's pbxproj.project file elementary structure.
+-- Reference: http://www.monobjc.net/xcode-project-file-format.html
+data PbxProj = PbxProj
+  { archiveVersion :: Text
+  , objectVersion :: Text
+  , rootObject :: Text
+  , classes :: Maybe AsciiValue
+  , objects :: Maybe AsciiValue
+  }
+  deriving (Show, Eq, Ord)
+
+(|->) :: AsciiValue -> Text -> Maybe AsciiValue
```

Comment:
Why did you make `|->` the operator and not something like `LookupTextFromAsciiDict`?

I don't love my suggestion either and I can see you wanting something concise, but unless there is a reason to use `|->` I think something a little more descriptive would be ideal.